src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -621,6 +621,14 @@ internal void SetRequestHeaders()
                     ThrowOOMIfFalse(Interop.Http.SListAppend(slist, NoTransferEncoding));
                 }
 
+                // Since libcurl adds an Expect header if it sees enough post data, we need to explicitly block
+                // it if caller specifically does not want to set the header
+                if (_requestMessage.Headers.ExpectContinue.HasValue &&
+                    !_requestMessage.Headers.ExpectContinue.Value)
```

Comment:
As a very minor perf improvement, this could be:
```C#
if (_requestMessage.Headers.ExpectContinue.GetValueOrDefault())
```
since both null and false should be ignored and only true is meaningful.